tests/unit/providers/test_mistral_provider.py

```
@@ -101,6 +101,97 @@ def test_patch_messages_with_multiple_valid_tool_calls() -> None:
     ]
 
 
+@pytest.mark.parametrize(
+    "reasoning",
+    ["thought hard", {"content": "thought hard"}],
+    ids=["serialized-string", "object-form"],
+)
+def test_patch_messages_rebuilds_thinking_chunk_for_replay(reasoning: Any) -> None:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `reasoning`

(ANN401)